rasa/utils/tensorflow/models.py

```diff
@@ -539,6 +548,84 @@ def __init__(
         # set up tf layers
         self._tf_layers: Dict[Text, tf.keras.layers.Layer] = {}
 
+    def adjust_sparse_layers(
+        self,
+        data_example: Dict[Text, Dict[Text, List[FeatureArray]]],
+        new_sparse_feature_sizes: Dict[Text, Dict[Text, List[int]]],
+        old_sparse_feature_sizes: Dict[Text, Dict[Text, List[int]]],
+    ) -> None:
+        """Adjusts sizes of `DenseForSparse` layers.
+
+        Updates sizes of `DenseForSparse` layers by comparing current sparse feature
+        sizes to old ones. This must be done before fine-tuning starts to account
+        for any change in the size of sparse features that might have happened
+        because of addition of new data. The function compiles the model, fits a
+        sample data on it to activate updated layer(s) and updates the data signatures.
+
+        New and old sparse feature sizes could look like this:
+        {TEXT: {FEATURE_TYPE_SEQUENCE: [4, 24, 128], FEATURE_TYPE_SENTENCE: [4, 128]}}
+
+        Args:
+            data_example: a data example that is stored with the ML component.
+            new_sparse_feature_sizes: sizes of current sparse features.
+            old_sparse_feature_sizes: sizes of sparse features the model was
+                                      previously trained on.
+        """
+        self._update_dense_for_sparse_layers(
+            new_sparse_feature_sizes, old_sparse_feature_sizes
+        )
+        self._compile_and_fit(data_example)
+
+    def _update_dense_for_sparse_layers(
+        self,
+        new_sparse_feature_sizes: Dict[Text, Dict[Text, List[int]]],
+        old_sparse_feature_sizes: Dict[Text, Dict[Text, List[int]]],
+    ) -> None:
+        """Updates `DenseForSparse` layers.
+
+        `if` condition is necessary because only `RasaCustomLayer` can adjust
+        sparse layers for incremental training by default.
```

Comment:
These kind of comments should go as in-line comments and not in the docstring. You are trying to comment on a particular line. Best place for it is just above the line.